### ISSUE TITLE
Changing line to work with pytrees in IterativeRefinement

### ISSUE THREAD

**Issue by FerranAlet** [google/jaxopt#311]
Hi,

I'm trying to use IterativeRefinement, but ran into

```
File ~/lib/python3.8/site-packages/jaxopt/_src/eq_qp.py", line 146, in matvec_regularized_qp                                                                                     
    return Stop + ridge * primal, Sbottom - ridge * dual_eq                                                          
TypeError: unsupported operand type(s) for *: 'float' and 'dict'  
```

I think it's because [this line](https://github.com/google/jaxopt/blob/main/jaxopt/_src/eq_qp.py#L146) should perform a tree_scalar_mul to be able to work with pytrees and not just arrays.

In particular, changing that line to
`tree_add(Stop, tree_scalar_mul(ridge, primal)), tree_sub(Sbottom, tree_scalar_mul(ridge, dual_eq)) ` made it work for me

**Comment by Algue-Rythme** [comment]
Hi Ferran

Good catch !

Some unit tests about this behavior are missing it seems ;)

**Comment by mblondel** [comment]
@FerranAlet or @Algue-Rythme a pull-request would be welcome!

**Comment by mblondel** [comment]
No, our preferred workflow is via github pull requests.

**Comment by Algue-Rythme** [comment]
Fixed by #345
